BlazarService/src/main/java/com/hubspot/blazar/util/SlackClientWrapper.java

```diff
@@ -0,0 +1,129 @@
+      metricRegistry.counter("successful-slack-channel-sends").inc();
+    } catch (Exception e) {
+      metricRegistry.counter("failed-slack-channel-sends").inc();
+      LOG.error("Could not send slack message {}", attachment, e);
+    }
+  }
+
+  public void sendMessageToUserByEmail(String email, String message, SlackAttachment attachment) {
+    Optional<SlackUser> user = Optional.fromNullable(session.findUserByEmail(email));
+    if (!user.isPresent()) {
+      LOG.error("Could not find user with email {} ", email);
+      return;
+    }
+    sendMessageToUser(user.get(), message, attachment);
+  }
+
+  public void sendMessageToUser(SlackUser user, String message, SlackAttachment attachment) {
+    try {
+      makeSlackMessageSendingRetryer().call(() -> sendMessage(user, message, attachment));
+      metricRegistry.counter("successful-slack-dm-sends").inc();
+    } catch (Exception e) {
+      LOG.error("Could not send slack message {}", attachment, e);
+      metricRegistry.counter("failed-slack-dm-sends").inc();
+    }
+
+  }
+
+  private boolean sendMessage(SlackUser user, String message, SlackAttachment attachment) {
+    Optional<SlackMessageHandle<SlackMessageReply>> result = Optional.fromNullable(session.sendMessageToUser(user, message, attachment));
+    if (!result.isPresent()) {
+      LOG.warn("Failed to send slack message to user: {} message: {} slack result was null", user.getRealName(), attachment.toString());
+      return false;
+    }
+
+    ParsedSlackReply reply = result.get().getReply();
+    if (!reply.isOk()) {
+      LOG.warn("Failed to send slack message to user: {} message: {} error: {}", user.getRealName(), attachment.toString(), reply.getErrorMessage());
+      return false;
+    }
+    return true;
+
+  }
+
+  private boolean sendMessage(SlackChannel channel, String message, SlackAttachment attachment) {
+    Optional<SlackMessageHandle<SlackMessageReply>> result = Optional.fromNullable(session.sendMessage(channel, message, attachment));
+
+    if (!result.isPresent()) {
+      LOG.warn("Failed to send slack message to channel: {} message: {} slack result was null", channel.getName(), attachment.toString());
+      return false;
+    }
+
+    ParsedSlackReply reply = result.get().getReply();
+    if (!reply.isOk()) {
+      if (reply.getErrorMessage().equals("not_in_channel")) {
+        // There is nothing the bot can do about this. It must be invited returning `true` here prevents sentries etc.
+        return true;
+      } else if (reply.getErrorMessage().equals("is_archived")) {
+        // ignore errors sending to archived channels
+        return true;
+      }
+
+      LOG.warn("Failed to send slack message to channel: {} message: {} error: {}", channel.getName(), attachment.toString(), reply.getErrorMessage());
+      return false;
+    }
+    return true;
+  }
+
+
+  private static Retryer<Boolean> makeSlackMessageSendingRetryer() {
+    return RetryerBuilder.<Boolean>newBuilder()
```

Comment:
The classes that are wrapped by the retrier don't have any try/catch. Are you sure that they are not going to throw any exceptions that your retrier will not handle